src/Plugin/EndpointCheck/Ping.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/**
+ * @file
+ * contains \Drupal\relaxed\Plugin\Endpoint\WorkspaceEndpoint
+ */
+
+namespace Drupal\relaxed\Plugin\EndpointCheck;
+
+use Drupal\relaxed\Entity\EndpointInterface;
+use Drupal\relaxed\Plugin\EndpointCheckBase;
+
+/**
+ * @EndpointCheck(
+ *   id = "ping",
+ *   label = "Ping endpoint"
+ * )
+ */
+Class Ping extends EndpointCheckBase {
+
+  public function __construct(array $configuration, $plugin_id, $plugin_definition) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->execute($configuration['endpoint']);
```

Comment:
I'd rather break this `execute()` method out to the interface, and make it the callers responsibility to call this method. Doing this in the constructor means we have no control over when to instatiate vs when to pass around plugin instances. It can make it very difficult to work with in a predicable way.